servers/quarkus-authn/src/main/java/org/projectnessie/server/authn/NessieAuthMechanism.java

```diff
@@ -0,0 +1,116 @@
+
+import io.quarkus.security.AuthenticationFailedException;
+import io.quarkus.security.identity.IdentityProviderManager;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.identity.request.AnonymousAuthenticationRequest;
+import io.quarkus.security.identity.request.AuthenticationRequest;
+import io.quarkus.vertx.http.runtime.security.ChallengeData;
+import io.quarkus.vertx.http.runtime.security.HttpAuthenticationMechanism;
+import io.quarkus.vertx.http.runtime.security.HttpCredentialTransport;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.projectnessie.quarkus.config.QuarkusNessieAuthenticationConfig;
+
+/**
+ * HTTP authentication mechanism implementation to disallow all unauthorized accesses, except for
+ * the explicitly configured paths and static resources.
+ */
+@ApplicationScoped
+public class NessieAuthMechanism implements HttpAuthenticationMechanism {
+
+  /**
+   * {@code HttpAuthenticationMechanism}s with a higher priority are "asked" first, let
+   * `NessieAuthMechanism` be called quite late, basically as the last one.
+   */
+  public static final int PRIORITY = 0;
+
+  private final boolean enabled;
+  private final AuthenticationFailedException unauthorized;
+  private final Set<String> anonymousPaths;
+  private final Set<String> anonymousPathPrefixes;
+
+  @Inject
+  public NessieAuthMechanism(QuarkusNessieAuthenticationConfig config) {
+    this.enabled = config.enabled();
+    this.unauthorized = new AuthenticationFailedException("Missing or unrecognized credentials");
+    Set<String> paths = new HashSet<>();
+    config.anonymousPaths().ifPresent(paths::addAll);
+    config.anonymousPathPrefixes().ifPresent(paths::addAll);
+    this.anonymousPaths = paths;
+    this.anonymousPathPrefixes =
+        config
+            .anonymousPathPrefixes()
+            .map(prefixes -> prefixes.stream().map(p -> p + '/').collect(Collectors.toSet()))
+            .orElse(Set.of());
+  }
+
+  boolean checkPath(String path) {
+    // Allow certain preconfigured paths (e.g. health checks) to be serviced without
+    // authentication.
+    return anonymousPaths.contains(path)
+        || anonymousPathPrefixes.stream().anyMatch(path::startsWith);
+  }
+
+  @Override
+  public int getPriority() {
+    return PRIORITY;
+  }
+
+  @Override
+  public Uni<SecurityIdentity> authenticate(
+      RoutingContext context, IdentityProviderManager identityProviderManager) {
+    if (enabled) {
+      var path = context.request().path();
+      if (checkPath(path)) {
+        return Uni.createFrom().nullItem();
+      }
+
+      if (!path.contains("..")) {
+        if (getClass().getClassLoader().getResource("META-INF/resources" + path) != null) {
```

Comment:
Sorry, I'm very confused why we need this special case here :thinking:

Reply:
@snazy : Could you comment with some examples for these resources and how accessing them worked before this PR?

Reply:
TBH, I'm skeptical about opening anonymous access to all jar-based resources.

Reply:
@dimas-b - this is how it worked before and how it's standard in Quarkus, right?

Reply:
@dimas-b what's the issue with allowing _static_ resources?
It does _not_ allow access to _any_ resource - only _static web resources_.

Reply:
If opens them by default, without a user action... which is not ideal from my POV. Previously we listed all anonymous resources explicitly. I'd prefer to keep the explicit whitelist.

However, if you have some new use cases for static resources, I do not have a strong objection... Maybe we can list that as a changelog item. WDYT?